### PR TITLE
Alec/tfac 365

### PR DESCRIPTION
**Pull request by amaliwanag2**

### CODE REVIEW

**Comment by amaliwanag2** [comment]
We can't really have the UI review step working properly until we have storybook on master b/c it wants to compare the diff against whats on master

also @kmjennison i accidentally branched off of the cats work so please review and merge the cat day notification first

File: .github/workflows/chromatic.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Workflow name
+name: 'Chromatic Deployment'
+
+# Event for the workflow
+on: push
```
Comment (amaliwanag2) [inline .github/workflows/chromatic.yml:5]: i was originally doing it on PR request, but I think on push is fine and that's what they recommend for some git history thing

File: .github/workflows/chromatic.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+          ref: ${{ github.head_ref }}
+          fetch-depth: 0
+      - run: yarn
+        #👇 Adds Chromatic as a step in the workflow
+      - uses: chromaui/action@v1
+        # Options required for Chromatic's GitHub Action
+        with:
+          projectToken: ${{ secrets.CHROMATIC_PROJECT_TOKEN }}
+          token: ${{ secrets.GITHUB_TOKEN }}
+          autoAcceptChanges: true
```
Comment (amaliwanag2) [inline .github/workflows/chromatic.yml:24]: for now, I don't want to deal with manually having to accept storybook changes in the storybook ui.  It also requires approval when you add new stories

File: package.json [hunk 3]
```diff
@@ -15,7 +15,10 @@
     "format:fix": "yarn run format --write",
     "lint": "eslint \"**/*.{js,jsx}\" --ignore-pattern \"next.config.*js\"",
     "relay": "relay-compiler --src ./src/ --artifactDirectory ./src/relay/__generated__ --exclude '**/__test__/**'  '**/__generated__/**' --schema ./src/schema/schema.graphql",
-    "find-unused": "echo \"This will find unused files. However, note that it ignores mocks, tests, and utils in the testHelpers directory.\n\" && next-unused"
+    "find-unused": "echo \"This will find unused files. However, note that it ignores mocks, tests, and utils in the testHelpers directory.\n\" && next-unused",
+    "storybook": "start-storybook -p 6006",
+    "build-storybook": "build-storybook",
+    "chromatic": "npx chromatic --auto-accept-changes"
```
Comment (amaliwanag2) [inline package.json:21]: you can manually deploy from local with this.  You'll need to set a local env variable tho

File: src/components/ImpactCounter.js [hunk 4]
```diff
@@ -95,8 +95,19 @@ const ImpactCounter = (props) => {
 }
 
 ImpactCounter.propTypes = {
+  /**
+   Checks if the number of treats should be included
+  */
```
Comment (amaliwanag2) [inline src/components/ImpactCounter.js:100]: storybook parses these comments and makes them the description of the prop

File: .storybook/main.js [hunk 5]
```diff
@@ -0,0 +1,4 @@
+module.exports = {
+  stories: ['../src/**/*.stories.mdx', '../src/**/*.stories.@(js|jsx|ts|tsx)'],
```
Comment (amaliwanag2) [inline .storybook/main.js:2]: tldr anything with a .stories. gets picked up

File: .storybook/main.js [hunk 6]
```diff
@@ -0,0 +1,4 @@
...
+  addons: ['@storybook/addon-links', '@storybook/addon-essentials'],
```
Comment (amaliwanag2) [inline .storybook/main.js:3]: these addons are for being able to manipulate props and for parsing and generating docs

File: .storybook/preview.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+  actions: { argTypesRegex: '^on[A-Z].*' },
+  controls: {
+    matchers: {
+      color: /(background|color)$/i,
+      date: /Date$/,
+    },
+  },
+}
+
+export const decorators = [
```
Comment (amaliwanag2) [inline .storybook/preview.js:13]: in general, storybook has a paradigm where it looks for specifically named exports in .stories files and in the .storybook/ folder.  Docs go over each of the named exports

File: package.json (hunk 3 above)
Comment (amaliwanag2) [reply package.json:21]: github action invokes build-storybook

File: .storybook/preview.js [hunk 8]
```diff
@@ -0,0 +1,19 @@
+import { ThemeProvider } from '@material-ui/core/styles'
+import theme from 'src/utils/theme'
+export const parameters = {
```
Comment (kmjennison) [inline .storybook/preview.js:3]: What are these?

File: package.json [hunk 9]
```diff
@@ -51,11 +54,17 @@
...
+    "@storybook/addon-essentials": "^6.3.6",
+    "@storybook/addon-links": "^6.3.6",
+    "@storybook/react": "^6.3.6",
     "@testing-library/react-hooks": "^4.0.0",
     "babel-jest": "^26.6.3",
+    "babel-loader": "^8.2.2",
     "babel-plugin-module-resolver": "^4.1.0",
     "babel-plugin-relay": "^10.1.2",
     "babel-plugin-transform-dynamic-import": "^2.1.0",
+    "chromatic": "^5.10.0-canary.0",
```
Comment (kmjennison) [inline package.json:67]: Any reason to use canary?

File: src/components/ImpactCounter.stories.jsx [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+const Template = (args) => <ImpactCounter {...args} />
+export const standard = Template.bind({})
+standard.args = {
+  includeNumber: true,
+  number: 10,
+  progress: 50,
+}
+
+export const halfFull = Template.bind({})
+halfFull.args = {
```
Comment (kmjennison) [inline src/components/ImpactCounter.stories.jsx:20]: NBD, but "Half full" is a weird name when "standard" is already half full.

File: src/__tests__/pages/index.test.js [hunk 11]
```diff
@@ -8,6 +8,7 @@ import { act } from 'react-dom/test-utils'
 import {
   showMockAchievements,
   showBackgroundImages,
+  showDevelopmentOnlyDemoPages,
```
Comment (kmjennison) [inline src/__tests__/pages/index.test.js:11]: Create a standalone feature flag for `showMissionsFeature`.

File: src/pages/index.js [hunk 12]
```diff
@@ -55,6 +57,7 @@ import { accountURL, achievementsURL, surveyLink } from 'src/utils/urls'
 import {
   showMockAchievements,
   showBackgroundImages,
+  showDevelopmentOnlyDemoPages,
```
Comment (kmjennison) [inline src/pages/index.js:60]: Same as above -- new env var.

File: src/components/MissionHubButton.js [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+      </Button>
+    </Link>
+  )
+}
+MissionHubButton.displayName = 'MissionHubButton'
+MissionHubButton.propTypes = {
+  /**
+    the status of the current mission
+  */
+  status: PropTypes.oneOf(['pending', 'started', 'completed', '']),
```
Comment (kmjennison) [inline src/components/MissionHubButton.js:59]: Empty string is unclear. "unstarted'?

File: src/components/MissionHubButton.stories.jsx [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+      default: 'black',
+      values: [
+        { name: 'grey', value: '#F2F2F2' },
+        { name: 'black', value: '#000000' },
+      ],
+    },
+  },
+}
+
+// eslint-disable-next-line react/jsx-props-no-spreading
```
Comment (kmjennison) [inline src/components/MissionHubButton.stories.jsx:19]: If you want, you can make an exception for this rule for `.stories.` files so we don't have to do this every time.. There's an example in .eslintrc.js for Jest files.

File: src/components/MissionHubButton.stories.jsx [hunk 15]
```diff
@@ -0,0 +1,34 @@
+import React from 'react'
+
+import MissionHubButton from './MissionHubButton'
+
+export default {
+  title: 'Components/MissionHubButton',
+  component: MissionHubButton,
+  parameters: {
+    backgrounds: {
```
Comment (kmjennison) [inline src/components/MissionHubButton.stories.jsx:9]: Are these parameters used?

**Review by kmjennison — CHANGES_REQUESTED** [review]
Some changes + some questions about Storybook.

File: .storybook/preview.js (hunk 8 above)
Comment (amaliwanag2) [reply .storybook/preview.js:3]: Parameters are a set of static, named metadata about a story, used to control the behavior of Storybook features and addons.  This is boiler plate to enable the "actions" addon, which records handlers being invoked

File: package.json (hunk 9 above)
Comment (amaliwanag2) [reply package.json:67]: no i just ran yarn chromatic.  Seems to be working just fine for now tho

File: src/components/MissionHubButton.stories.jsx (hunk 15 above)
Comment (amaliwanag2) [reply src/components/MissionHubButton.stories.jsx:9]: yeah, just allows you to change the background of the canvas

**Review by kmjennison — APPROVED** [review]
LGTM! 🦊

**Pull request merged**